src/librustc/mir/interpret/value.rs

```diff
@@ -272,11 +272,13 @@ impl<'tcx, Tag> Scalar<Tag> {
 
     #[inline]
     pub fn from_bool(b: bool) -> Self {
+        // Guaranteed to be truncated and does not need sign extension.
```

Comment:
These seem like they could be separate? But fine to leave them in too.

Reply:
I had to add `from_i32` for the Miri side of this. So while I was at it I figured I might as well add the comments here, instead of making them a separate PR.